src/docs/product/issues/issue-details/performance-issues/db-main-thread-io.mdx

```diff
@@ -0,0 +1,50 @@
+---
+title: "Database on Main Thread"
+sidebar_order: 20
+description: "Learn more about Database on Main Thread and how to diagnose and fix these issues."
+---
+
+The main UI thread in a mobile application handles user interface events such as button presses and page scrolls. To prevent App Hangs and Application Not Responding errors, the main UI thread should not be used for performing long-running operations like Database queries. These kinds of operations block the whole UI until they finish running and get in the way of the user interacting with the app.
```

Comment:
```suggestion
The main UI thread in a mobile application handles user interface events such as button presses and page scrolls. To prevent App Hangs and Application Not Responding errors, the main UI thread should not be used for performing long-running operations like database queries. These kinds of operations block the whole UI until they finish running and get in the way of the user interacting with the app.
```